### ISSUE TITLE
dist/npm missing schemas (and other babel compile woes)

### ISSUE THREAD

**Issue by sublimator** [ripple/ripple-lib#367]
Few issues:
* es6 runtime patches (like String.prototype.endsWith) seem to be missing
* npm compile before each script run is slow

When you try to run `npm run-script perf` it errors, when it can't find the `schemas/` folder. On top of that, if it's there (I sym linked it), there's another error due to a missing endsWith function. I looked that up, and it's apparently a String method only added in ES6.

It's also quite painful having to wait for the `npm run-script compile` each time (On my machine it takes about 5 seconds) and the logs are noisy. Apparently you can use `babel-node` which seems to do on the fly compiling (and String.endsWith seems available) in what feels like around a second.

You of course need to update the scripts from:
```javascript
var Amount = require('../dist/npm').Amount;
```
to:
```javascript
var Amount = require('../src').Amount;
```

I'm guessing there's probably other ES6 issues with the missing runtime?

**Comment by clark800** [comment]
So I just learned that babel doesn't transpile all es6 features in `runtime` mode unless you additionally require the polyfill. This seems to contradict the documentation, so I filed an issue with Babel on GitHub. In the meantime, I just wrote a function for `endsWith` and fixed the perf test in this pull request: https://github.com/ripple/ripple-lib/pull/370

We can't use babel-node or the babel require hook for production because we have to support users who are running es5 without babel. But npm test currently does run with the require hook. That's something that we should probably change because we aren't exactly testing the production code.

I haven't figured out how to use caching yet, but that would be a good way to speed it up. I added the following JIRA tasks: https://ripplelabs.atlassian.net/browse/RLJS-386, https://ripplelabs.atlassian.net/browse/RLJS-387
